### PR TITLE
Merge 2.3 to master

### PR DESCRIPTION
**Pull request by juho-jaakkola**

### CODE REVIEW

**Comment by juho-jaakkola** [comment]
Getting this out of the way, so merging translations will be cleaner.

**Comment by jdalsem** [comment]
is this PR right? feels like too much of the file is touched

**Comment by juho-jaakkola** [comment]
It did seem a bit weird. Could be that I didn't resolved the conflicts correctly.

**Comment by mrclay** [comment]
Here are the conflicts: https://github.com/Elgg/Elgg/pull/11579/conflicts

File: engine/classes/Elgg/Database/Seeds/Seed.php [hunk 1]
```diff
@@ -35,4 +39,643 @@ public function __construct($limit = null) {
...
+			/* @var $user ElggUser */
+
+			elgg_set_user_validation_status($guid, $this->faker->boolean(), 'seeder');
+
+			$user->setNotificationSetting('email', false);
+			$user->setNotificationSetting('site', true);
+
+			$profile_fields = elgg_get_config('profile_fields');
+
+			$user = $this->populateMetadata($user, $profile_fields, $metadata);
```
Comment (mrclay) [inline engine/classes/Elgg/Database/Seeds/Seed.php]: Nope. Needs populateAnnotations. I vaguely remember writing this.

File: engine/classes/Elgg/Database/Seeds/Seed.php [hunk 2]
```diff
@@ -35,4 +39,643 @@ public function __construct($limit = null) {
...
+			$attributes['subtype'] = strtolower($this->faker->word);
+		}
+
+		if (empty($metadata['tags'])) {
+			$metadata['tags'] = $this->faker->words(10);
+		}
+
+		if (empty($attributes['owner_guid'])) {
+			if ($container instanceof ElggGroup) {
+				$members = elgg_get_entities_from_relationship([
```
Comment (mrclay) [inline engine/classes/Elgg/Database/Seeds/Seed.php]: elgg_get_entities now, right?

File: engine/classes/Elgg/Database/Seeds/Seed.php [hunk 3]
```diff
@@ -35,4 +39,643 @@ public function __construct($limit = null) {
...
+		}
+
+		if (empty($attributes['access_id'])) {
+			$attributes['access_id'] = $this->getRandomAccessId($owner, $container);
+		}
+
+		$object = false;
+
+		try {
+			$class = get_subtype_class('object', $attributes['subtype']);
```
Comment (mrclay) [inline engine/classes/Elgg/Database/Seeds/Seed.php]: Needs rewrite...

File: engine/classes/Elgg/Database/Seeds/Seed.php [hunk 4]
```diff
@@ -35,4 +39,643 @@ public function __construct($limit = null) {
...
+
+		$success = 0;
+
+		if (!$limit) {
+			$limit = $this->faker->numberBetween(1, 20);
+		}
+
+		while ($success < $limit) {
+
+			$comment = new ElggObject();
```
Comment (mrclay) [inline engine/classes/Elgg/Database/Seeds/Seed.php]: new ElggComment :)

**Comment by mrclay** [comment]
If we're desperate to merge I'd be OK with temporarily disabling seeding, doing this merge without the rewriting, then adding an issue to fix issues in the seeder.

**Comment by juho-jaakkola** [comment]
I don't think there's any immediate rush.

But can someone else handle the conflicts? I'm not familiar with this feature at all.

**Comment by mrclay** [comment]
Hmm, so how do I search my... `git branch | wc -l` ...798 local branches for "populateAttributes"? 😢

**Comment by mrclay** [comment]
With `git rev-list --all | xargs git grep -F populateAnnotations` I found it! I'll see what I can do here.

**Comment by mrclay** [comment]
My changes can be pulled in here: https://github.com/juho-jaakkola/Elgg/pull/3

**Comment by juho-jaakkola** [comment]
I've added Steve's modifications to the PR.

**Comment by hypeJunction** [comment]
This looks weird. I'd say ignore all 2.3 changes on that class.

**Comment by hypeJunction** [comment]
Seeding has been rewritten in master to use traits. So, just trashing the 2.3 class is the way to go. https://github.com/Elgg/Elgg/pull/11586

**Comment by jdalsem** [comment]
done in #11586

**Pull request closed without merging**